src/librustc_data_structures/bitvec.rs

```diff
@@ -293,15 +293,23 @@ impl<R: Idx, C: Idx> SparseBitMatrix<R, C> {
     ///
     /// Returns true if this changed the matrix, and false otherwise.
     pub fn add(&mut self, row: R, column: C) -> bool {
-        self.vector[row].insert(column)
+        if let None = self.vector.get(row) {
+            self.vector.push(SparseBitSet::new());
+        }
+
+        if let Some(row) = self.vector.get_mut(row) {
```

Comment:
Now this can be `self.vector[row].insert(column)` as before

Reply:
Fixed this.